pkg/testutil/test_suite_runner.go

```diff
@@ -0,0 +1,202 @@
+	t := runner.Delegate.GoTestRunner()
+	t.Run(scenarioName, func(t *testing.T) {
+		rm := fixtureCxt.resourceManager
+		assert := assert.New(t)
+
+		var actual acktypes.AWSResource = nil
+		var err error = nil
+		switch unitUnderTest {
+		case "ReadOne":
+			actual, err = rm.ReadOne(context.Background(), fixtureCxt.desired)
+		case "Create":
+			actual, err = rm.Create(context.Background(), fixtureCxt.desired)
+		case "Update":
+			delta := runner.Delegate.ResourceDescriptor().Delta(fixtureCxt.desired, fixtureCxt.latest)
+			actual, err = rm.Update(context.Background(), fixtureCxt.desired, fixtureCxt.latest, delta)
+		case "Delete":
+			actual, err = rm.Delete(context.Background(), fixtureCxt.desired)
+		default:
+			panic(errors.New(fmt.Sprintf("unit under test: %s not supported", unitUnderTest)))
+		}
+		runner.assertExpectations(assert, expectation, actual, err)
+	})
+}
+
+/* assertExpectations validates the actual outcome against the expected outcome.
+There are two components to the expected outcome, corresponding to the return values of the resource manager's CRUD operation:
+	1) the actual return value of type AWSResource ("expect.latest_state" in test_suite.yaml)
+	2) the error ("expect.error" in test_suite.yaml)
+With each of these components, there are three possibilities in test_suite.yaml, which are interpreted as follows:
+	1) the key does not exist, or was provided with no value: no explicit expectations, don't assert anything
+	2) the key was provided with value "nil": explicit expectation; assert that the error or return value is nil
+	3) the key was provided with value other than "nil": explicit expectation; assert that the value matches the
+		expected value
+However, if neither expect.latest_state nor error are provided, assertExpectations will fail the test case.
+*/
+func (runner *TestSuiteRunner) assertExpectations(assert *assert.Assertions, expectation *Expect, actual acktypes.AWSResource, err error) {
+	if expectation.LatestState == "" && expectation.Error == "" {
+		fmt.Println("Invalid test case: no expectation given for either latest_state or error")
+		assert.True(false)
+		return
+	}
+
+	// expectation exists for at least one of LatestState and Error; assert results independently
+	if expectation.LatestState == "nil" {
+		assert.Nil(actual)
+	} else if expectation.LatestState != "" {
+		expectedLatest := runner.loadAWSResource(expectation.LatestState)
+		assert.NotNil(actual)
+
+		delta := runner.Delegate.ResourceDescriptor().Delta(expectedLatest, actual)
+		assert.Equal(0, len(delta.Differences))
+		if len(delta.Differences) > 0 {
+			fmt.Println("Unexpected differences:")
+			for _, difference := range delta.Differences {
+				fmt.Printf("Path: %v, expected: %v, actual: %v\n", difference.Path, difference.A, difference.B)
+			}
+		}
+
+		// Check that the yaml files are equivalent.
```

Comment:
can you also add the reason why we are doing this because one might get confused as to why we need it apart from `delta` and `Delegate.Equal`

Something like `// Check that the yaml files are equivalent. This makes it easier to make changes to unit test cases`

Also can you move this line to the end below the Delegate.Equal

Reply:
I made the comment change you mentioned.

However I kept the YamlEqual call before the Equal call because if the Equal call is first and the Equal assert fails (something in the status doesn't match) then the test stops/fails right there and you are not able to see the YamlEqual output which is more useful for debugging.